graphx/src/main/scala/org/apache/spark/graphx/impl/EdgeRDDImpl.scala

```diff
@@ -87,7 +87,7 @@ class EdgeRDDImpl[ED: ClassTag, VD: ClassTag] private[graphx] (
 
   /** The number of edges in the RDD. */
   override def count(): Long = {
-    partitionsRDD.map(_._2.size.toLong).reduce(_ + _)
+    partitionsRDD.map(_._2.size.toLong).fold(0)(_ + _)
```

Comment:
The code below has the same problem in case of an empty edge?
https://github.com/apache/spark/blob/ae5b2a6a92be4986ef5b8062d7fb59318cff6430/graphx/src/main/scala/org/apache/spark/graphx/lib/SVDPlusPlus.scala#L75

Reply:
Oh nice catch, yeah. I searched for other "reduce" calls but didn't hit this one.